indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -546,6 +553,41 @@ public TaskStatus runTask(final TaskToolbox toolbox)
     }
   }
 
+  private void emitUnparseableEvents(TaskToolbox toolbox)
+  {
+    String taskId = getId();
```

Comment:
There are inherent similarities between these two methods here and in ParallelIndexSupervisorTask. Would be good if we could move the commonalities to an utils or a common class

Reply:
makes sense